meal-sharing-master/src/backend/api/meals.js

```diff
@@ -4,15 +4,15 @@ const { as } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
-//GET	Returns all meals 
+//GET	Returns all meals  
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
+    // knex syntax for selecting things. Look up the documentation for knex for further info 
     const titles = await knex("meals").select("title");
     const meals = await knex("meals");
```

Comment:
It is not desirable to read all meal rows from the DB and then do the filtering in Node (image the performance if there were >100k meals in the DB). Instead, the idea is that you should construct a single SQL query that will apply different filters based on the query parameters.

Reply:
sorry for late response, I can't figure out what I have to do here? could you give me example

Reply:
If we use the contacts example DB, then it could look something like this:


```js
const knex = require('knex')({
  client: 'mysql2',
  connection: {
    host: '127.0.0.1',
    user: 'root',
    password: 'my-secret-pw',
    database: 'contacts'
  }
});

let query = knex('contacts')
query = query.where({ first_name: 'Orhan' })
query = query.where({ last_name: 'Toy' })
console.log('query:', query.toString())
query.then(result => console.log('result', result))
```

The idea is then to check some conditions and depending on the conditions, add certain extra filters to the query.

If you understand what I mean and manage to do it, that's great but it's not mandatory to get this homework approved.

Reply:
i guess you mean like that :
 if ("maxPrice" in request.query) {
      const maxPrice = parseInt(request.query.maxPrice);
      if (isNaN(maxPrice)) {
        return response
          .status(400)
          .send({ error: "Max Price must be integers" });
      }
      filteredMeals = await knex("meals").where("price", "<=", maxPrice);
    }

Reply:
But wouldn't you still only support one query parameter at a time then?